src/Components/DataHistoryExplorer/Internal/TimeSeriesViewer/Internal/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -0,0 +1,124 @@
+import React, {
+    useCallback,
+    useContext,
+    useEffect,
+    useMemo,
+    useRef,
+    useState
+} from 'react';
+import {
+    ITimeSeriesChartProps,
+    ITimeSeriesChartStyleProps,
+    ITimeSeriesChartStyles
+} from './TimeSeriesChart.types';
+import { getStyles } from './TimeSeriesChart.styles';
+import { classNamesFunction, useTheme, styled } from '@fluentui/react';
+import HighChartsWrapper from '../../../../../HighChartsWrapper/HighChartsWrapper';
+import ChartCommandBar from './Internal/ChartCommandBar/ChartCommandBar';
+import { IHighChartSeriesData } from '../../../../../HighChartsWrapper/HighChartsWrapper.types';
+import { transformADXTimeSeriesToHighChartsSeries } from '../../../../../../Models/SharedUtils/DataHistoryUtils';
+import { DataHistoryExplorerContext } from '../../../../DataHistoryExplorer';
+import { useTimeSeriesData } from '../../../../../../Models/Hooks/useTimeSeriesData';
+import { usePrevious } from '@fluentui/react-hooks';
+import { IDataHistoryChartOptions } from '../../../../../../Models/Types/Generated/3DScenesConfiguration-v1.0.0';
+import { QuickTimeSpanKey } from '../../../../../../Models/Constants/Enums';
+import { QuickTimeSpans } from '../../../../../../Models/Constants/Constants';
+
+const getClassNames = classNamesFunction<
+    ITimeSeriesChartStyleProps,
+    ITimeSeriesChartStyles
+>();
+
+const TimeSeriesChart: React.FC<ITimeSeriesChartProps> = (props) => {
+    const { timeSeriesTwinList, styles } = props;
+
+    // state
+    const [chartOptions, setChartOptions] = useState<IDataHistoryChartOptions>({
+        yAxisType: 'independent',
+        defaultQuickTimeSpanInMillis:
+            QuickTimeSpans[QuickTimeSpanKey.Last15Mins],
+        aggregationType: 'avg'
+    });
+
+    // hooks
+    const { adapter } = useContext(DataHistoryExplorerContext);
+    const xMinDateInMillisRef = useRef<number>(null);
+    const xMaxDateInMillisRef = useRef<number>(null);
+    const {
+        query,
+        deeplink,
+        data,
+        fetchTimeSeriesData,
+        isLoading
+    } = useTimeSeriesData({
+        adapter,
+        connection: adapter.getADXConnectionInformation(),
+        quickTimeSpanInMillis: chartOptions.defaultQuickTimeSpanInMillis,
+        twins: timeSeriesTwinList
+    });
+    const highChartSeriesData: Array<IHighChartSeriesData> = useMemo(
+        () =>
+            timeSeriesTwinList?.length
+                ? transformADXTimeSeriesToHighChartsSeries(
+                      data,
+                      timeSeriesTwinList
+                  )
+                : [],
+        [data, timeSeriesTwinList]
+    );
+
+    // callbacks
+    const updateXMinAndMax = useCallback(() => {
+        const nowInMillis = Date.now();
+        xMinDateInMillisRef.current =
```

Comment:
no necessarily since the Highcharts would render again with its **highChartSeriesData** prop changes anyhow, but when it does I just need a x values value passed along with updated data to draw the x axis values correctly in the chart. So it is not a single and individual variable that is needed to be part of state to re-render but a variable passed along with other props which forces to re-render